cmd/gardenlet/app/options.go

```diff
@@ -72,6 +73,11 @@ func (o *options) complete() error {
 }
 
 func (o *options) validate() error {
+	// Add feature flags
+	if err := features.DefaultFeatureGate.SetFromMap(o.config.FeatureGates); err != nil {
+		return err
+	}
```

Comment:
Why was this moved to `validate()`? This sounds wrong - nothing is validated here. I would suggest to move it back. Now it's also inconsistent compared to how the feature gates are handled in the other binaries.

Reply:
I described the problem in the other comment. TLDR; during the validation of the seed object inside the config, we check if the feature gate is enabled or not.